src/test/run-pass/generator/size-moved-locals.rs

```diff
@@ -55,8 +55,20 @@ fn overlap_move_points() -> impl Generator<Yield = (), Return = ()> {
     }
 }
 
+fn overlap_x_and_y() -> impl Generator<Yield = (), Return = ()>{
+    static || {
+        let x = [0u8; 1024];
+        yield;
+        drop(x);
+        let y = [0u8;1024];
+        yield;
+        drop(y);
+    }
+}
+
 fn main() {
     assert_eq!(1028, std::mem::size_of_val(&move_before_yield()));
     assert_eq!(1032, std::mem::size_of_val(&move_before_yield_with_noop()));
     assert_eq!(2056, std::mem::size_of_val(&overlap_move_points()));
+    assert_eq!(2052, std::mem::size_of_val(&overlap_x_and_y()));
```

Comment:
So, the value I would expect to see for this test is closer to 1024, if x and y are indeed being overlapped.

The reason x and y are _not_ overlapped in this test is: static arrays are `Copy`, so calling `drop()` on them drops a copy of the array instead of the array itself. (This is unfortunate, and should be fixed by a pass that removes unnecessary copies from MIR.)

The way to fix this test is to use `Foo` instead of raw arrays, like in the above tests. (Also, please use `FOO_SIZE` in the test instead of 1024 for consistency.)

Reply:
Opened #62952 to track the issue with missed optimizations for locals which are `Copy`.

Reply:
I see, fixed tests.